src/test/scala/chiselTests/experimental/ProbeSpec.scala

```diff
@@ -0,0 +1,94 @@
+package chiselTests.experimental
+
+import chisel3._
+import chisel3.experimental.Probe
+import chiselTests.{ChiselFlatSpec, Utils}
+import circt.stage.ChiselStage
+
+class ProbeSpec extends ChiselFlatSpec with Utils {
+  "U-Turn example" should "emit FIRRTL probe statements and expressions" in {
+    val chirrtl = ChiselStage.emitCHIRRTL(
+      new Module {
+
+        val io = IO(new Bundle {
+          val x = Input(UInt(1.W))
+          val y = Output(UInt(1.W))
+        })
+
+        class UTurn() extends RawModule {
+          val io = IO(new Bundle {
+            val in = Input(Probe(Bool()))
+            val out = Output(Probe(Bool()))
+          })
+          io.out := io.in
+        }
+
+        val u1 = Module(new UTurn())
+        val u2 = Module(new UTurn())
+
+        val n = Probe.probe(io.x)
+
+        Probe.define(u1.io.in, n)
+        Probe.define(u2.io.in, u1.io.out)
+
+        io.y := Probe.read(u2.io.out)
+
+        Probe.forceInitial(u1.io.out, false.B)
+
+        Probe.releaseInitial(u1.io.out)
+
+        Probe.force(clock, io.x, u2.io.out, u1.io.out)
+        Probe.release(clock, io.y, u2.io.out)
+      },
+      Array("--full-stacktrace")
+    )
+
+    chirrtl should include("output io : { flip in : Probe<UInt<1>>, out : Probe<UInt<1>>}")
+    chirrtl should include("define u1.io.in = probe(io.x)")
+    chirrtl should include("define u2.io.in = u1.io.out")
+    chirrtl should include("io.y <= read(u2.io.out)")
+    chirrtl should include("force_initial(u1.io.out, UInt<1>(\"h0\")")
+    chirrtl should include("release_initial(u1.io.out")
+    chirrtl should include("force(clock, io.x, u2.io.out, u1.io.out)")
+    chirrtl should include("release(clock, io.y, u2.io.out)")
+  }
+
+  "Connectors" should "work with probes" in {
```

Comment:
@azidar does this test have the output you are expecting?

Reply:
I'm not sure if this is the question, but lets say I have a Parent and a Child that both have `FooBundle`. Can I do parent.io :<>= child.io, and it works like one expects to forward the probe through? does it error, or emit firrtl that won't do what we want?

Reply:
ah i missed that your example already has a bunch of nested FooBundle.